core/domain/suggestion_services.py

```diff
@@ -1826,3 +1828,805 @@ def update_question_suggestion(
+    content_word_count = len(content_plain_text.split())
+
+    translation_review_stat_model = (
+        suggestion_models.TranslationReviewStatsModel.get(
+            suggestion.change.language_code, user_id, topic_id
+        ))
+
+    if translation_review_stat_model is None:
+        suggestion_models.TranslationReviewStatsModel.create(
+            language_code=suggestion.change.language_code,
+            reviewer_user_id=user_id,
+            topic_id=topic_id,
+            reviewed_translations_count=1,
+            reviewed_translation_word_count=content_word_count,
+            accepted_translations_count=1,
+            accepted_translations_with_reviewer_edits_count=1 * int(
+                suggestion.edited_by_reviewer),
+            accepted_translation_word_count=content_word_count * int(
+                is_accepted),
+            first_contribution_date=(
+                datetime.datetime.strptime(
+                    str(
+                        suggestion.last_updated.utcnow().date().isoformat()),
+                        '%Y-%m-%d'
+                    ).date()),
+            last_contribution_date=(
+                datetime.datetime.strptime(
+                    str(
+                        suggestion.last_updated.utcnow().date().isoformat()),
+                        '%Y-%m-%d'
+                    ).date())
+        )
+    else:
+        translation_review_stat = (
+            _create_translation_review_stats_from_model(
+                translation_review_stat_model))
+
+        translation_review_stat.reviewed_translations_count += 1
+        translation_review_stat.reviewed_translation_word_count += (
+            content_word_count)
+        translation_review_stat.accepted_translations_count += 1 * int(
+            is_accepted)
+        (
+            translation_review_stat
+            .accepted_translations_with_reviewer_edits_count
+        ) += 1 + int(suggestion.edited_by_reviewer)
+        translation_review_stat.last_contribution_date = (
+                datetime.datetime.strptime(
+                    str(
+                        suggestion.last_updated.utcnow().date().isoformat()),
+                        '%Y-%m-%d'
+                    ).date())
+
+        _update_translation_review_stats_models([translation_review_stat])
+
+    update_translation_contribution_stats_at_review(
+        suggestion, suggestion.author_id, target_id)
+
+
+def update_question_contribution_stats_at_submission(
+    suggestion: suggestion_registry.BaseSuggestion,
+    user_id: str, target_id: str) -> None:
+    """Creates/updates QuestionContributionStatsModel model for given question
+    submitter when a question is submitted.
+
+    Args:
+        suggestion: Suggestion. The suggestion domain object that is being
+            submitted.
+        user_id: str. The ID of the corresponding submitter.
+        target_id: str. The ID of the target entity being suggested to.
+    """
+    topic_ids = []
```

Comment:
In Python, it's more idiomatic to use list comprehension for something like this, e.g.:

```
topic_ids = [
  topic.topic_id
  for topic in skill_services.get_all_topic_assignments_for_skill(target_id)
]
```

Reply:
Done